### PR TITLE
Users support

### PR DESCRIPTION
**Pull request by tjrivera**
This assumes #19 is close to being merged. I imagine this would be reviewed post-merge of that patch. There are changes there specifically around EE testing that I believe are worth keeping. 

I added TODOs here around integration with other Kong endpoints that I anticipate including in subsequent patches. (Roles, Permissions, etc.)

### CODE REVIEW

**Comment by hbagdi** [comment]
Please rebase this on the latest `main` branch.

**Comment by tjrivera** [comment]
@hbagdi rebase complete.

File: kong/kong.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type Client struct {
 	Targets        *TargetService
 	Workspaces     *WorkspaceService
 	Admins         *AdminService
+	Users          *UserService
```
Comment (hbagdi) [inline kong/kong.go]: Please rename Users to RBACUsers. Same applies to UserService.
User is generic enough to cause a confusion here.

**Review by hbagdi — COMMENTED** [review]
Looks good other than one change.

**Comment by tjrivera** [comment]
@hbagdi I'll make these requested changes. I imagine it makes sense to reopen this against the new Kong org repo?

**Comment by hbagdi** [comment]
Yes please.

**Comment by tjrivera** [comment]
Moved to https://github.com/Kong/go-kong/pull/1

**Pull request closed without merging**